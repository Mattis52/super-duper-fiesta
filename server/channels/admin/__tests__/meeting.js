jest.mock('../../../models/meeting');
const { endGAM, toggleRegistration } = require('../meeting');
const { getGenfors, getActiveGenfors, updateGenfors } = require('../../../models/meeting');
const { generateSocket, generateGenfors, generateUser } = require('../../../utils/generateTestData');
const permissionLevels = require('../../../../common/auth/permissions');

describe('toggleRegistration', () => {
  beforeEach(() => {
    getActiveGenfors.mockImplementation(async () => generateGenfors());
    updateGenfors.mockImplementation(async (genfors, data) =>
      ({ ...genfors, ...data, pin: genfors.pin }));
  });

  const generateData = data => (Object.assign({
    registrationOpen: true,
  }, data));

  it('emits a toggle action that closes registration', async () => {
    const socket = generateSocket();
    await toggleRegistration(socket, generateData());

    expect(socket.emit.mock.calls).toMatchSnapshot();
    expect(socket.broadcast.emit.mock.calls).toMatchSnapshot();
  });

  it('emits a toggle action that opens registration', async () => {
    const socket = generateSocket();
    await toggleRegistration(socket, generateData({ registrationOpen: false }));

    expect(socket.emit.mock.calls).toMatchSnapshot();
    expect(socket.broadcast.emit.mock.calls).toMatchSnapshot();
  });
});

describe('endGAM', () => {
  beforeAll(() => {
    getActiveGenfors.mockImplementation(async () => generateGenfors());
    updateGenfors.mockImplementation(async (genfors, data) =>
      generateGenfors({ ...genfors, ...data, pin: genfors.pin }));
  });

  it('closes meeting if requested', async () => {
    const socket = generateSocket(generateUser({ permissions: permissionLevels.IS_MANAGER }));
    const genfors = generateGenfors();
    getActiveGenfors.mockImplementation(async () => genfors);
    getGenfors.mockImplementation(async () => genfors);

    await endGAM(socket);

    expect(socket.emit.mock.calls).toMatchSnapshot();
    expect(socket.broadcast.emit.mock.calls).toMatchSnapshot();
  });
});
