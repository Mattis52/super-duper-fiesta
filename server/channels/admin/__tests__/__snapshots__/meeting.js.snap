// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggleRegistration emits a toggle action that closes registration 1`] = `
Array [
  Array [
    "action",
    Object {
      "data": Object {
        "id": "1",
        "pin": 3141592653,
        "registrationOpen": false,
      },
      "type": "TOGGLED_REGISTRATION_STATE",
    },
  ],
]
`;

exports[`toggleRegistration emits a toggle action that closes registration 2`] = `
Array [
  Array [
    "action",
    Object {
      "data": Object {
        "id": "1",
        "pin": 3141592653,
        "registrationOpen": false,
      },
      "type": "TOGGLED_REGISTRATION_STATE",
    },
  ],
]
`;

exports[`toggleRegistration emits a toggle action that opens registration 1`] = `
Array [
  Array [
    "action",
    Object {
      "data": Object {
        "id": "1",
        "pin": 3141592653,
        "registrationOpen": true,
      },
      "type": "TOGGLED_REGISTRATION_STATE",
    },
  ],
]
`;

exports[`toggleRegistration emits a toggle action that opens registration 2`] = `
Array [
  Array [
    "action",
    Object {
      "data": Object {
        "id": "1",
        "pin": 3141592653,
        "registrationOpen": true,
      },
      "type": "TOGGLED_REGISTRATION_STATE",
    },
  ],
]
`;
